src/nbl/asset/ISPIRVOptimizer.cpp

```diff
@@ -0,0 +1,85 @@
+#include "nbl/asset/ISPIRVOptimizer.h"
+
+#include "spirv-tools/optimizer.hpp" 
+#include "spirv-tools/libspirv.hpp"
+
+#include "nbl/core/core.h"
+#include "nbl/core/IReferenceCounted.h"
+#include "os.h"
+
+using namespace nbl::asset;
+
+static constexpr spv_target_env SPIRV_VERSION = spv_target_env::SPV_ENV_UNIVERSAL_1_5;
+
+nbl::core::smart_refctd_ptr<ICPUBuffer> ISPIRVOptimizer::optimize(const uint32_t* _spirv, uint32_t _dwordCount) const
+{
+    spvtools::Optimizer opt(SPIRV_VERSION);
+
+    // lots of these taken from https://www.lunarg.com/wp-content/uploads/2020/05/SPIR-V-Shader-Legalization-and-Size-Reduction-Using-spirv-opt_v1.2.pdf
+    opt .RegisterPass(spvtools::CreateMergeReturnPass())
```

Comment:
there should be an engine ENUM to list all these.

then the constructor of the optimizer should take an ENUM initializer list of these and the `spvtools::Optimizer` should be constructed there and kept. 

For parallelism we assume access to `ISPIRVOptimizer` is externally synchronised, this means that either user makes sure different optimizer is used for each thread concurrently, or that we `clone()` the default optimizer whenever we use the asset manager's or driver's default optimizer (make access to it a const).